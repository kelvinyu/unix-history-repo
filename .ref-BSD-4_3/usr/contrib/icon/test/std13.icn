record array(a,b,c,d,e,f,g)

procedure dummy(u,v,x,y,z)
   suspend u | v
   return x
end

procedure main()
   write("real(2) ----> ",image(real(2)) | "none")
   write("real(2.0) ----> ",image(real(2.0)) | "none")
   write("real(2.7) ----> ",image(real(2.7)) | "none")
   write("real(\"2\") ----> ",image(real("2")) | "none")
   write("real(\" 2\") ----> ",image(real(" 2")) | "none")
   write("real(\"2 \") ----> ",image(real("2 ")) | "none")
   write("real(\"+2\") ----> ",image(real("+2")) | "none")
   write("real(\"-2\") ----> ",image(real("-2")) | "none")
   write("real(\"- 2\") ----> ",image(real("- 2")) | "none")
   write("real(\" -    2 \") ----> ",image(real(" -    2 ")) | "none")
   write("real(\"\") ----> ",image(real("")) | "none")
   write("real(\"--2\") ----> ",image(real("--2")) | "none")
   write("real(\" \") ----> ",image(real(" ")) | "none")
   write("real(\"-\") ----> ",image(real("-")) | "none")
   write("real(\"+\") ----> ",image(real("+")) | "none")
   write("real(\".\") ----> ",image(real(".")) | "none")
   write("real(\".3\") ----> ",image(real(".3")) | "none")
   write("real(\"0.3\") ----> ",image(real("0.3")) | "none")
   write("real(\" . 3\") ----> ",image(real(" . 3")) | "none")
   write("real(\"e2\") ----> ",image(real("e2")) | "none")
   write("real(\"22222222222222222222222222222222222222222222222222222222222\") ----> ",image(real("22222222222222222222222222222222222222222222222222222222222")) | "none")
   write("real(\"3e500\") ----> ",image(real("3e500")) | "none")
   write("real(\"7r4\") ----> ",image(real("7r4")) | "none")
   write("real(\"4r7\") ----> ",image(real("4r7")) | "none")
   write("real(\"4r 7\") ----> ",image(real("4r 7")) | "none")
   write("real(\"7r 4\") ----> ",image(real("7r 4")) | "none")
   write("real(\"16rff\") ----> ",image(real("16rff")) | "none")
   write("real(\"36rcat\") ----> ",image(real("36rcat")) | "none")
   write("real(\"36Rcat\") ----> ",image(real("36Rcat")) | "none")
   write("real(\"36rCAT\") ----> ",image(real("36rCAT")) | "none")
   write("real(\"1r1\") ----> ",image(real("1r1")) | "none")
   write("real(\"2r0\") ----> ",image(real("2r0")) | "none")
   write("16rff / 0.0 ----> ",image(16rff / 0.0) | "none")
end
