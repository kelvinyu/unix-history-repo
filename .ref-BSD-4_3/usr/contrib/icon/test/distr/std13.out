real(2) ----> 2.0
real(2.0) ----> 2.0
real(2.7) ----> 2.7
real("2") ----> 2.0
real(" 2") ----> 2.0
real("2 ") ----> 2.0
real("+2") ----> 2.0
real("-2") ----> -2.0
real("- 2") ----> none
real(" -    2 ") ----> none
real("") ----> none
real("--2") ----> none
real(" ") ----> none
real("-") ----> none
real("+") ----> none
real(".") ----> none
real(".3") ----> 0.3
real("0.3") ----> 0.3
real(" . 3") ----> none
real("e2") ----> none
real("22222222222222222222222222222222222222222222222222222222222") ----> none
real("3e500") ----> none
real("7r4") ----> 4.0
real("4r7") ----> none
real("4r 7") ----> none
real("7r 4") ----> none
real("16rff") ----> 255.0
real("36rcat") ----> 15941.0
real("36Rcat") ----> 15941.0
real("36rCAT") ----> 15941.0
real("1r1") ----> none
real("2r0") ----> 0.0
Run-time error 204 at line 41 in std13.icn
real overflow, underflow, or division by zero
